### ISSUE TITLE
Line of Sight GP tool needs checks for valid input parameters

### ISSUE THREAD

**Issue by kgonzago** [Esri/military-tools-geoprocessing-toolbox#302]
### Task to be done 16 May 2018
- [x] Verify tool does not work properly in ArcMap and Pro
- [x] Add validation in the execution step
- [x] Verify automated tests for no observers and targets
- [x] Verify Manually
- [x] Verify in Military Tools Build
https://github.com/Esri/military-tools-geoprocessing-toolbox/pull/313

Line of Sight GP tool needs to have checks in place to determine if input parameters are valid for observers and targets.  

## Expected Behavior

After testing in Pro, the tool can be executed without supplying either an observer or target input parameter. The tool will eventually fail and occasionally will crash ArcGIS Pro entirely.

## Current Behavior

Currently, the tool accepts zero to many observers and targets. The tool should accept **one** to many observers and targets.

## Possible Solution

A valid check for observer and target input parameters need to implemented. This will be part of the tool validation process.

## Steps to Reproduce (for bugs)

1. Open Pro
2. Add a DEM
3. Launch Line of Sight tool
4. Add Observers
5. Do not add Targets
6. Execute tool

## Context

This issue was found when testing the Line of Sight tool on another issue. 

## Your Environment

* Version used: Pro 2.1
* Environment name and version (e.g. Chrome 39, node.js 5.4): N/A
* Operating System and version (desktop or mobile): Win10
* Link to your project:

**Comment by kgonzago** [comment]
Issue fixed in this [PR](https://github.com/Esri/military-tools-geoprocessing-toolbox/pull/310)

**Comment by kgonzago** [comment]
Test cases were added and are reflected in this [PR](https://github.com/Esri/military-tools-geoprocessing-toolbox/pull/312)

**Comment by topowright** [comment]
I am looking at the pull request for this issue. 
https://github.com/Esri/military-tools-geoprocessing-toolbox/pull/313

**Comment by dfoll** [comment]
tested this on @kgonzago   "kg/#302" branch and got the expected results from the fix which is for the tool to run and fail with a very clear error message to "Please provide at least one observer" or "Please provide at least one target" if there is an observer but no target.  Cannot test from build

**Comment by topowright** [comment]
Needs to be tested from the build. Waiting on the build to be finished.

**Comment by dfoll** [comment]
testing in both Pro and ArcMap

**Comment by kgonzago** [comment]
@dfoll Parameters were incorrect. Fixed it in this [PR](https://github.com/Esri/military-tools-geoprocessing-toolbox/pull/332)

**Comment by dfoll** [comment]
verified in the .pyt with MT build 40
